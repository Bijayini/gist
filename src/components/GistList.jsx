import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

import theme from '../lib/theme';
import BodyCopy from '../ui-components/BodyCopy';

import TagList from './TagList';
import ForkList from './ForkList';

const GistList = ({ gistDetails }) => {
  return (
    <Fragment>
      <div className="gist-list-container">
        {gistDetails.map(gist => (
          <div className="gist-list" key={gist.id}>
            <div className="id">
              <BodyCopy variant="bold" xs="2" xm="2" xl="2" color="textColor">
                {gist.id}
              </BodyCopy>
            </div>
            {!!(gist.tags && gist.tags.length) && (
              <div className="tags">
                <TagList tags={gist.tags} />
              </div>
            )}
            {!!(gist.forks && gist.forks.length) && (
              <div className="forks">
                <ForkList forks={gist.forks} />
              </div>
            )}
          </div>
        ))}
      </div>
      <style jsx="true">
        {`
          .gist-list-container {
            margin-top: 20px;
            border: 1px solid ${theme.border};
            border-radius: 3px;
            border-bottom: none;
          }
          .gist-list {
            border-bottom: 1px solid ${theme.border};
            display: flex;
            padding: 10px;
          }
          .id {
            width: 40%;
          }
          .tags {
            width: 30%;
          }
          .forks {
            width: 30%;
          }
        `}
      </style>
    </Fragment>
  );
};

GistList.propTypes = {
  gistDetails: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      tags: PropTypes.arrayOf(PropTypes.string).isRequired,
      forks: PropTypes.arrayOf(PropTypes.string).isRequired,
    }),
  ).isRequired,
};
export default GistList;
