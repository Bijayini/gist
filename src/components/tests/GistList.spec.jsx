import React from 'react';
import { shallow } from 'enzyme';

import GistList from '../GistList';

describe('<GistList />', () => {
  let props;
  let component;

  beforeEach(() => {
    props = {
      gistDetails: [
        {
          id: 'some id',
          forks: [],
          tags: ['some tag'],
        },
        {
          id: 'some id2',
          forks: ['some url1'],
          tags: [],
        },
      ],
    };
    component = shallow(<GistList {...props} />);
  });

  it('should match with snapshot', () => {
    expect(component.debug()).toMatchSnapshot();
  });
});
