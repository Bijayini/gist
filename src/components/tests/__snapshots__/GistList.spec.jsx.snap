// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GistList /> should match with snapshot 1`] = `
"<Fragment>
  <div className=\\"jsx-569556867 gist-list-container\\">
    <div className=\\"jsx-569556867 gist-list\\">
      <div className=\\"jsx-569556867 id\\">
        <BodyCopy variant=\\"bold\\" xs=\\"2\\" xm=\\"2\\" xl=\\"2\\" color=\\"textColor\\" className=\\"\\" uppercase={false} styles={false}>
          some id
        </BodyCopy>
      </div>
      <div className=\\"jsx-569556867 tags\\">
        <TagList tags={{...}} />
      </div>
    </div>
    <div className=\\"jsx-569556867 gist-list\\">
      <div className=\\"jsx-569556867 id\\">
        <BodyCopy variant=\\"bold\\" xs=\\"2\\" xm=\\"2\\" xl=\\"2\\" color=\\"textColor\\" className=\\"\\" uppercase={false} styles={false}>
          some id2
        </BodyCopy>
      </div>
      <div className=\\"jsx-569556867 forks\\">
        <ForkList forks={{...}} />
      </div>
    </div>
  </div>
  <JSXStyle id=\\"1096698853\\" dynamic={{...}}>
    .gist-list-container.__jsx-style-dynamic-selector{margin-top:20px;border:1px solid #c8c8c8;border-radius:3px;border-bottom:none;}.gist-list.__jsx-style-dynamic-selector{border-bottom:1px solid #c8c8c8;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding:10px;}.id.__jsx-style-dynamic-selector{width:40%;}.tags.__jsx-style-dynamic-selector{width:30%;}.forks.__jsx-style-dynamic-selector{width:30%;}
  </JSXStyle>
</Fragment>"
`;
